xarray/core/common.py

```diff
@@ -539,24 +543,69 @@ def resample(self, freq, dim, how='mean', skipna=None, closed=None,
         """
         from .dataarray import DataArray
 
+        if dim is not None:
+            return self._resample_immediately(freq, dim, how, skipna, closed,
+                                              label, base, keep_attrs)
+
```

Comment:
Should we explicitly disallow the specification of `how` when a `indexers` is provided? I don't think we need to support that functionality since we know it will be deprecated.